service/kvmclusterconfig/framework.go

```diff
@@ -0,0 +1,117 @@
+package kvmclusterconfig
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/informer"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/cluster-operator/service/kvmclusterconfig/v1"
+)
+
+type FrameworkConfig struct {
+	G8sClient    versioned.Interface
+	K8sClient    kubernetes.Interface
+	K8sExtClient apiextensionsclient.Interface
+	Logger       micrologger.Logger
+
+	// Name is the name of the project.
+	Name string
+}
+
+func NewFramework(config FrameworkConfig) (*framework.Framework, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.G8sClient must not be empty")
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.K8sExtClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sExtClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	var err error
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.DefaultConfig()
+
+		c.K8sExtClient = config.K8sExtClient
+		c.Logger = config.Logger
+
+		crdClient, err = k8scrdclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
+
+		c.Watcher = config.G8sClient.CoreV1alpha1().KVMClusterConfigs("")
+
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var v1ResourceSet *framework.ResourceSet
+	{
+		c := v1.ResourceSetConfig{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+
+			HandledVersionBundles: []string{
+				"0.1.0",
+			},
+			Name: config.Name,
+		}
+
+		v1ResourceSet, err = v1.NewResourceSet(c)
```

Comment:
I don't really have strong opinions on one way or the other, but I think TIm's argument on current style was good.